jax/tree_util.py

```diff
@@ -247,7 +274,21 @@ def _replace_nones(sentinel, tree):
       return tree
 
 no_initializer = object()
-def tree_reduce(function, tree, initializer=no_initializer):
+
+@overload
```

Comment:
Oddly enough the annotations on `tree_reduce` are blocking me from submitting this PR. They reveal a type error in some code that uses `tree_reduce`. I'm corresponding with the author if it's a real bug or a type checker problem. I think however I could submit this right now if we temporarily removed the two overloads on `tree_reduce`.

Reply:
Okay, as you wish!